tasklist/client/e2e/utils/waitForAssertion.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+async function waitForAssertion(options: {
+  assertion: () => Promise<void>;
+  onFailure: () => Promise<void>;
+  maxRetries?: number;
+}) {
+  const {assertion, onFailure: fallback, maxRetries = 3} = options;
```

Comment:
I am not familiar with playwright but in Optimize, we use retries for tests to eliminate the flaky tests.
I see that playright have something similar, maybe it is worth using if it is not used already.
https://playwright.dev/docs/test-retries

Reply:
In this case the Playwright retries don't work because we don't have polling here so we need to reload the page.